### DISCUSSION TITLE
[Discussion] Will native views that are currently embedded in platform-specific projects be supported in MAUI?

### DISCUSSION THREAD

**Discussion by BurkusCat** [dotnet/maui#57] [General]
With the new project structure, I was curious if native views like this: https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/native-views/xaml will be supported.

I currently have embedded the XAML + C# directly from the [Windows Community Toolkit for the InfiniteCanvas](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/master/Microsoft.Toolkit.Uwp.UI.Controls/InfiniteCanvas) control. I have done this so I can heavily customise it; I am not just using the UWP NuGet package.
![image](https://user-images.githubusercontent.com/7528322/82436178-511ad280-9a8d-11ea-8141-b7f5f9d9029f.png)

In my shared project, I have a View.
![image](https://user-images.githubusercontent.com/7528322/82436909-73612000-9a8e-11ea-8ef0-a4a42ffc4916.png)

In my UWP project I have a custom renderer that uses the native UWP control for that View (Android and iOS just have renderers that are placeholders). In my UWP specific `App.xaml` I have had to include the XAML for the control:
![image](https://user-images.githubusercontent.com/7528322/82436496-d3a39200-9a8d-11ea-931b-ca9fca82f3c6.png)

In videos, I didn't see any UWP `App.xaml` or android manifest etc. Could something like this be accomplished in MAUI or will this functionality be dropped?

**Comment by dhindrik** [comment]
As I understand it, the "one project thing" is powers by multi-targeting. So you should be able to do the same thing now, but instead of having it in a separate project you will add conditions to it.

**Comment by davidortinau** [comment]
@BurkusCat yes, you’ll still be able to do this! 

We are still working through the scenarios we need to address for Single Project, and this is a great one for us to keep in mind.

.NET MAUI will still work with multi-project solutions as it does today.

If anyone else has similar usage today, I would love to see your examples too! I’ll leave this open for discussion.

**Comment by xleon** [comment]
Please make some way to avoid pre-processor directives. Having `#if` everywhere is horrible.

**Comment by mattleibow** [comment]
I answered a similar question on SO about this. Not quite the "stick anything in the XAML", but allows for custom controls to be added: https://stackoverflow.com/a/73514661/1117914

**Comment by alenny811** [comment]
The way it was made in Xamarin-xaml was fantastic:
[https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/native-views/xaml](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/native-views/xaml)

Hope to see the same elegancy in MAUI some day

**Comment by mcyenikoylu** [comment]
I am waiting for this too. Although Windows operating system is dominant in enterprise level application development, my customers use Android and iOS devices in the factory and for their mobile personnel in the field. As a single developer, I am doing research to solve everything with MAUI. I want to move my WinUI project to MAUI.
